client/app/scripts/liveblog-advertising/controllers/advertising.js

```diff
@@ -0,0 +1,241 @@
+            type: $scope.advertType.name,
+            text: freetypeService.htmlContent($scope.advertType.template, $scope.freetypesData),
+            meta: {data: $scope.freetypesData}
+        }
+        $scope.dialogAdvertLoading = true;
+
+        api('advertisements').save($scope.advert, newAd)
+        .then(handleAdvertSaveSuccess, handleAdvertSaveError);
+    }
+
+    loadAdverts();
+
+
+    //COLLECTIONS
+    function loadCollections(silent) {
+        silent = silent || false;
+        if (!silent) {
+            $scope.collectionsLoading = true;
+        }
+        return api('collections').query({where: {deleted: false}})
+        .then(function(data) {
+            $scope.collections = data._items;
+            if (!silent) {
+                notify.info(gettext('Collections loaded'));
+            }
+            $scope.collectionsLoading = false;
+        })
+        .catch(function(data) {
+            $scope.collectionsLoading = false;
+            notify.error(gettext('There was an error getting the adverts'));
+        })
+    }
+
+    function handleCollectionSaveSuccess() {
+        notify.info(gettext('Collection saved successfully'));
+        $scope.collection = {};
+        $scope.collectionModalActive = false;
+        $scope.dialogCollectionLoading = false;
+        return loadCollections(true);
+    }
+
+    function handleCollectionSaveError() {
+        notify.error(gettext('Something went wrong, please try again later!'), 5000)
+    }
+
+    $scope.openCollectionDialog = function(collection) {
+        collection = collection || false;
+        // load all available adverts without showing any messages
+        loadAdverts(true).then(function() {
+            if (collection) {
+                // editing collection
+                $scope.collection = angular.copy(collection);
+                $scope.collection.checkAdverts = {};
+                //console.log('$scope.collection ', $scope.collection);
+                angular.forEach($scope.adverts, function(advert) {
+                    if ($scope.collectionHasAdvert($scope.collection, advert)) {
+                        $scope.collection.checkAdverts[advert._id] = true;
+                    } else {
+                        $scope.collection.checkAdverts[advert._id] = false;
+                    }
+                });
+            } else {
+                $scope.collection = {};
+                // for checkboxes and advert collections
+                $scope.collection.checkAdverts = {};
+                angular.forEach($scope.adverts, function(advert) {
+                    $scope.collection.checkAdverts[advert._id] = false;
+                });
+            }
+            $scope.collectionModalActive = true;
+        });
+    }
+
+    $scope.saveCollection = function() {
+
+        //create the saveable advertisement array for the collection
+        var advertisements = [];
+        angular.forEach($scope.collection.checkAdverts, function(checked, ad_id) {
```

Comment:
So you just use this array of booleans to create a new array? Why not just create the new array without creating an array of booleans?